File: Sources/CountriesFlagsObjC/source/UIImage+Bangladesh.m
//
//  UIImage+Bangladesh.m
//  CountriesFlags
//

#import "UIImage+Bangladesh.h"

@implementation UIImage (Bangladesh)

+(UIImage * _Nullable) flagBangladeshWithSize:(CGSize)size {
    UIColor* colorRed = [UIColor colorWithRed:244.0f/255.0f
                                        green:42.0f/255.0f
                                         blue:65.0f/255.0f
                                        alpha:1.0f];
    UIColor* colorBottleGreen = [UIColor colorWithRed:0.0f
                                                green:106.0f/255.0f
                                                 blue:78.0f/255.0f
                                                alpha:1.0f];
    
    UIGraphicsBeginImageContextWithOptions(size, NO, 0.0f);
    
    [colorBottleGreen setFill];
    UIRectFill(CGRectMake(0.0f, 0.0f, size.width, size.height));
    
    [colorRed setFill];
    CGFloat radius = size.height * 4.0f / 5.0f;
    UIBezierPath* path = [UIBezierPath bezierPathWithOvalInRect:CGRectMake(size.width * 2.0 / 5.0 - radius / 2.0,
                                                                           size.height / 2.0 - radius / 2.0,
                                                                           radius,
                                                                           radius)];
    [path fill];
    
    UIImage* image = UIGraphicsGetImageFromCurrentImageContext();
    UIGraphicsEndImageContext();
    
    return image;
}

@end

File: Sources/CountriesFlags/Countries/UIImage+Bangladesh.swift
//
//  UIImage+Bangladesh.swift
//  CountriesFlags
//

#if canImport(UIKit)

import UIKit

extension UIImage {
    /**
     [wikipedia]: https://en.wikipedia.org/wiki/Flag_of_Bangladesh "wikipedia"
     
     Generates the flag of Bangladesh.
     For more information see [wikipedia].
     
     - Parameters:
        - size: image size.
     
     - Returns: Returns the flag of Bangladesh.
     */
    internal static func flagBangladesh(with size: CGSize) -> UIImage? {
        let colorRed = UIColor(red: 244.0 / 255.0,
                               green: 42.0 / 255.0,
                               blue: 65.0 / 255.0,
                               alpha: 1.0)
        let colorBottleGreen = UIColor(red: 0.0,
                                       green: 106.0 / 255.0,
                                       blue: 78.0 / 255.0,
                                       alpha: 1.0)

        UIGraphicsBeginImageContextWithOptions(size, false, 0.0)

        colorBottleGreen.setFill()
        UIRectFill(CGRect(origin: .zero, size: size))

        colorRed.setFill()
        UIBezierPath(arcCenter: CGPoint(x: size.width * 2.0 / 5.0, y: size.height / 2.0),
                     radius: size.height * 4.0 / 10.0,
                     startAngle: 0.0,
                     endAngle: .pi * 2.0,
                     clockwise: true).fill()

        let image = UIGraphicsGetImageFromCurrentImageContext()
        UIGraphicsEndImageContext()

        return image
    }
}

#endif
